hw/ip/otbn/dv/uvm/tests/otbn_base_test.sv

```
@@ -14,12 +16,22 @@ class otbn_base_test extends dv_base_test #(
   // otbn_env_cfg: cfg
   // otbn_env:     env
 
-  virtual function void build_phase(uvm_phase phase);
+  function void build_phase(uvm_phase phase);
     super.build_phase(phase);
     cfg.has_ral = 1'b0;
+
+    cfg.mem_util = OtbnMemUtilMake("tb.dut");
+    `DV_CHECK_FATAL(cfg.mem_util != null);
+
+    if (!$value$plusargs("otbn_elf_dir=%0s", cfg.elf_dir)) begin
```

Please match the plusarg var name with the SV class variable name so that we can easily find it (I will update our coding style guide soon to indicate this).

---

Plusargs are generally override values set over the command line (`elf_dir` is a rare case where a default value cannot be set in the SV code). Hence, in most cases, we ignore the return value of `$value$plusargs`. This unfortunately has a side-effect - any typos in setting the plusarg will get silently ignored (example `+foo_bar=1` should have really been `+foobar=1`). To avoid the debug overhead due to situations like these, it is recommended that the plusarg name and the variable name be identical, so that correlation is intuitive and there are less chances of errors.

---

Ok, I've pushed up a version that changes the variable name to `cfg.otbn_elf_dir`.